### PR TITLE
osdocs397 update AWS permissions

### PR DESCRIPTION
**Pull request by kalexand-rh**
This is straight from https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/aws/permissions.go

https://jira.coreos.com/browse/OSDOCS-397

@wking, will you please check the list titles? 

built draft: http://file.rdu.redhat.com/kalexand/050919/osdocs397/installing/installing_aws/installing-aws-account.html#installation-aws-permissions-installing-aws-account

### CODE REVIEW

File: modules/installation-aws-permissions.adoc [hunk 1]
```diff
@@ -9,287 +9,172 @@ When you attach the `AdministratorAccess` policy to the IAM user that you create
...
+* `ec2:CreateRoute`
+* `ec2:CreateRouteTable`
+* `ec2:CreateSecurityGroup`
+* `ec2:CreateSubnet`
+* `ec2:CreateTags`
+* `ec2:CreateVpc`
+* `ec2:CreateVpcEndpoint`
+* `ec2:CreateVolume`
+* `ec2:DeleteSnapshot`
+* `ec2:DeregisterImage`
```
Comment (wking) [inline modules/installation-aws-permissions.adoc]: These last two are currently only used by `destroy cluster`.  We could use them to deregister our encrypted AMI copy once the bootstrap and control-plane machines came up, but we don't do that now.  So probably move them down into the *All permissions that are required to uninstall a cluster* section, at least for now.

File: modules/installation-aws-permissions.adoc [hunk 2]
```diff
@@ -9,287 +9,172 @@ When you attach the `AdministratorAccess` policy to the IAM user that you create
...
+* `ec2:DescribeVpcClassicLink`
+* `ec2:DescribeVpcClassicLinkDnsSupport`
+* `ec2:ModifyInstanceAttribute`
+* `ec2:ModifySubnetAttribute`
+* `ec2:ModifyVpcAttribute`
+* `ec2:RevokeSecurityGroupEgress`
+* `ec2:RunInstances`
+* `ec2:TerminateInstances`
+* `ec2:DeleteDhcpOptions`
+* `ec2:DeleteRoute`
```
Comment (wking) [inline modules/installation-aws-permissions.adoc]: These last two are only needed for `destroy cluster`.

File: modules/installation-aws-permissions.adoc [hunk 3]
```diff
@@ -9,287 +9,172 @@ When you attach the `AdministratorAccess` policy to the IAM user that you create
...
+* `ec2:RevokeSecurityGroupIngress`
+* `ec2:DisassociateRouteTable`
```
Comment (wking) [inline modules/installation-aws-permissions.adoc]: I *think* this is only needed for `destroy cluster`.

File: modules/installation-aws-permissions.adoc [hunk 4]
```diff
@@ -9,287 +9,172 @@ When you attach the `AdministratorAccess` policy to the IAM user that you create
...
+* `ec2:ReplaceRouteTableAssociation`
+* `ec2:DeleteRouteTable`
+* `ec2:DeleteSubnet`
```
Comment (wking) [inline modules/installation-aws-permissions.adoc]: These last two are only needed for `destroy cluster`.

File: modules/installation-aws-permissions.adoc [hunk 5]
```diff
@@ -9,287 +9,172 @@ When you attach the `AdministratorAccess` policy to the IAM user that you create
...
+* `ec2:DescribeNetworkInterfaces`
+* `ec2:ModifyNetworkInterfaceAttribute`
+* `ec2:DeleteNatGateway`
+* `ec2:DeleteSecurityGroup`
+* `ec2:DetachInternetGateway`
+* `ec2:DeleteInternetGateway`
+* `ec2:ReleaseAddress`
+* `ec2:DeleteVpc`
```
Comment (wking) [inline modules/installation-aws-permissions.adoc]: These last six (`DeleteNatGateway` and later) are only needed for `destroy cluster`.

File: modules/installation-aws-permissions.adoc [hunk 6]
```diff
@@ -9,287 +9,172 @@ When you attach the `AdministratorAccess` policy to the IAM user that you create
...
+.Required Elasticloadbalancing permissions for installation
+* `elasticloadbalancing:AddTags`
+* `elasticloadbalancing:ApplySecurityGroupsToLoadBalancer`
+* `elasticloadbalancing:AttachLoadBalancerToSubnets`
+* `elasticloadbalancing:CreateListener`
+* `elasticloadbalancing:CreateLoadBalancer`
+* `elasticloadbalancing:CreateLoadBalancerListeners`
+* `elasticloadbalancing:CreateTargetGroup`
+* `elasticloadbalancing:ConfigureHealthCheck`
+* `elasticloadbalancing:DeleteLoadBalancer`
```
Comment (wking) [inline modules/installation-aws-permissions.adoc]: This is only needed for `destroy cluster`.  Although the Kubernetes cloud provider must need this too to manage LoadBalancer Services.  Maybe ask the Cloud team how they get creds for that.

File: modules/installation-aws-permissions.adoc [hunk 7]
```diff
@@ -9,287 +9,172 @@ When you attach the `AdministratorAccess` policy to the IAM user that you create
...
+* `iam:SimulatePrincipalPolicy`
+* `iam:TagRole`
+
+.Required Route53 permissions for installation
+* `route53:ChangeResourceRecordSets`
+* `route53:ChangeTagsForResource`
+* `route53:GetChange`
+* `route53:GetHostedZone`
+* `route53:CreateHostedZone`
+* `route53:DeleteHostedZone`
```
Comment (wking) [inline modules/installation-aws-permissions.adoc]: This is only needed for `destroy cluster`.

File: modules/installation-aws-permissions.adoc [hunk 8]
```diff
@@ -9,287 +9,172 @@ When you attach the `AdministratorAccess` policy to the IAM user that you create
...
+* `s3:PutObject`
+* `s3:PutObjectAcl`
+* `s3:PutObjectTagging`
+* `s3:GetObject`
+* `s3:GetObjectAcl`
+* `s3:GetObjectTagging`
+* `s3:GetObjectVersion`
+* `s3:DeleteObject`
+
+.All permissions that are required to uninstall a cluster
```
Comment (wking) [inline modules/installation-aws-permissions.adoc]: This should be "All *additional* permissions..." or some such.  You need some of the above permissions as well, and we don't currently distinguish between install-time and destroy-time permission requirements in the installer code-base (although we probably should).

**Comment by kalexand-rh** [comment]
Thank you @wking! If you make any improvements or changes to this list in the code base, will you please let me know?

**Comment by kalexand-rh** [comment]
@jianlinliu, will you PTAL or suggest a different reviewer?

**Comment by jianlinliu** [comment]
LGTM.

**Comment by kalexand-rh** [comment]
Thank you both! I'm going to merge this. @wking, if you change the way you chunk the permissions in the installer, please let me know so I can update this topic.

**Pull request merged**